scipy/stats/_mannwhitneyu.py

```diff
@@ -0,0 +1,367 @@
+
+        statistic : float
+            The Mann-Whitney U statistic corresponding with sample `x`. See
+            Notes for the test statistic corresponding with sample `y`.
+        pvalue : float
+            The associated *p*-value for the chosen `alternative`.
+
+    Notes
+    -----
+    If ``U1`` is the statistic corresponding with sample `x`, then the
+    statistic corresponding with sample `y` is
+    `U2 = `x.shape[axis] * y.shape[axis] - U1``.
+
+    `mannwhitneyu2` is for independent samples. For related samples,
+    consider `scipy.stats.wilcoxon`.
+
+    `method` ``'exact'`` is recommended when there are no ties and when either
+    sample size is less than 8 [1]_. The implementation follows the recurrence
+    relation originally proposed in [1]_ as it is described in [3]_.
+    Note that the exact method is *not* corrected for ties, but
+    `mannwhitneyu2` will not raise errors or warnings if there are ties in the
+    data.
+
+    See Also
+    --------
+    scipy.stats.wilcoxon
+
+    References
+    ----------
+    .. [1] H.B. Mann and D.R. Whitney, "On a test of whether one of two random
+           variables is stochastically larger than the other", The Annals of
+           Mathematical Statistics, Vol. 18, pp. 50-60, 1947.
+    .. [2] Mann-Whitney U Test, Wikipedia,
+           http://en.wikipedia.org/wiki/Mann-Whitney_U_test
+    .. [3] A. Di Bucchianico, "Combinatorics, computer algebra, and the
+           Wilcoxon-Mann-Whitney test", Journal of Statistical Planning and
+           Inference, Vol. 79, pp. 349-364, 1999.
+    .. [4] Rosie Shier, "Statistics: 2.3 The Mann-Whitney U Test", Mathematics
+           Learning Support Centre, 2004.
+
+    Examples
+    --------
+    We follow the example from [4]_: nine randomly sampled young adults were
+    diagnosed with type II diabetes at the ages below.
+
+    >>> males = [19, 22, 16, 29, 24]
+    >>> females = [20, 11, 17, 12]
+
+    We use the Mann-Whitney U test to assess whether there is a statistically
+    significant difference in the age of diagnosis of males and females.
+    The null hypothesis is that the probability of a male diagnosis age being
+    greater than a female diagnosis age is equal to the probability of a
+    female diagnosis age being greater than a male diagnosis age. We decide
+    that a confidence level of 95% is required to reject the null hypothesis
+    in favor of the alternative: that either the diagnosis age of males is
+    stochastically greater than the diagnosis age of females or the diagnosis
+    age of females is stochastically greater than the diagnosis age of males.
+    Since the number of samples is very small and there are no ties in the
+    data, we can compare the test statistic against the *exact* distribution
+    of the test statistic under the null hypothesis.
+
+    >>> from scipy.stats import mannwhitneyu2
+    >>> U1, p = mannwhitneyu2(males, females, method="exact")
+    >>> print(U1)
```

Comment:
is print even needed or would >>> U1 also work?

Reply:
It is if you want output and these commands are in a script rather than run individually in the console.